src/init/features/functions/index.ts

```diff
@@ -30,6 +22,104 @@ export async function doSetup(setup: any, config: any, options: Options) {
       ensure(projectId, "runtimeconfig.googleapis.com", "unused", true),
     ]);
   }
+  setup.functions = {};
+
+  if (!config.src.functions) {
+    // if functions have not been initialized yet
+    setup.config.functions = [];
+    return codebaseSetup(setup, config, false);
+  } else {
+    // if functions have already been initialized
+    // make sure config is validated and normalized
+    setup.config.functions = normalizeAndValidate(setup.config.functions);
+
+    logger.info();
+    logger.info("Detected " + clc.bold("existing codebase(s)."));
+    logger.info();
+
+    const choices = [
+      {
+        name: "Initialize",
+        value: "new",
+      },
+      {
+        name: "Re-initialize",
+        value: "reinit",
+      },
+    ];
+    const initOpt = await promptOnce({
+      type: "list",
+      message: "Would you like to initialize a new codebase, or reinitialize an existing one?",
+      default: "new",
+      choices,
+    });
+    return codebaseSetup(setup, config, initOpt === "reinit");
+  }
+}
+
+/**
+ *  User dialogue to set up configuration for functions codebase.
+ */
+async function codebaseSetup(setup: any, config: Config, reinit: boolean): Promise<any> {
+  if (reinit) {
+    const choices = setup.config.functions.map((cbconfig: any) => {
+      return {
+        name: cbconfig["codebase"],
+        value: cbconfig["codebase"],
+      };
+    });
```

Comment:
It looks like we have an opportunity to make the flow smarter - if there is only 1 codebase, we shouldn't need the user to select from our choices.

Reply:
I wondered about that and decided against handling the single-codebase case in the name of "code-conciseness" (but really I was just being lazy). But agreed